cpmpy/tools/tune_solver.py

```diff
@@ -115,3 +115,60 @@ def _params_to_np(self,combos):
     def _np_to_params(self,arr):
         return {key: val for key, val in zip(self._param_order, arr)}
 
+
+
+class GridSearchTuner(ParameterTuner):
+
+    def __init__(self, solvername, model, all_params=None, defaults=None):
+        super().__init__(solvername, model, all_params, defaults)
+
+    def tune(self, time_limit=None, max_tries=None, fix_params={}):
+        """
+            :param: time_limit: Time budget to run tuner in seconds. Solver will be interrupted when time budget is exceeded
+            :param: max_tries: Maximum number of configurations to test
+            :param: fix_params: Non-default parameters to run solvers with.
+        """
+        if time_limit is not None:
+            start_time = time.time()
+
+        # Init solver
+        solver = SolverLookup.get(self.solvername, self.model)
+        solver.solve(**self.best_params)
+
+
+        self.base_runtime = solver.status().runtime
+        self.best_runtime = self.base_runtime
+
+        # Add default's runtime as first entry in configs
+        combos = list(param_combinations(self.all_params))
+
+        i = 0
+        if max_tries is None:
+            max_tries = len(combos)
+        while len(combos) and i < max_tries:
```

Comment:
i dont really like the config_idx below and this 'while', it should just be a simple for? If you want randomized order, maybe something like `for i, param_dict in rand.shuffle(param_combinations(self.all_params)):`?

Reply:
Ah yes, this tuner is indeed much simpler to code, updated